tools/oxide-bar/src/main.rs

```diff
@@ -197,45 +212,67 @@ impl OxideBar {
         );
     }
 
-    fn draw(&mut self, state: OxideState) {
+    fn draw(&mut self) {
         let cr = cairo::Context::new(self.cairo_surface.as_ref().unwrap())
             .expect("failed to create cairo context");
-        if self.composite_mgr {
+
+        if self.composite_mgr && self.config.color_bg.is_alpha {
             cr.set_operator(cairo::Operator::Source);
-            cr.set_source_rgba(0.9, 1.0, 0.9, 0.5);
+            info!("Using composite manager, disabling background color");
+            let (r, g, b, a) = self.config.color_bg.rgba();
+            cr.set_source_rgba(r, g, b, a);
         } else {
-            cr.set_source_rgb(0.9, 1.0, 0.9);
+            let (r, g, b) = self.config.color_bg.rgb();
+            cr.set_source_rgb(r, g, b);
         }
+
+        println!("Drawing background {:?}", self.config.color_bg);
         cr.paint().unwrap();
+
         if self.composite_mgr {
             cr.set_operator(cairo::Operator::Over);
         }
-        cr.show_text("Hi there").unwrap();
 
-        let ws_vec = state.get_workspace_list(self.screen);
+        //let (r, g, b) = self.config.color_txt.rgb();
+        //cr.set_source_rgb(r, g, b);
+        //let layout = create_layout(&cr).unwrap();
+        //println!("pixel size: {:?}", layout.pixel_size());
+        //layout.set_ellipsize(EllipsizeMode::End);
+        //layout.set_text("T");
+        //show_layout(&cr, &layout);
```

Comment:
is this code still needed ?:)

Reply:
if this is  okay i would like to keep it. This would be the foundation for future modules. But I could extract this into an example